### ISSUE TITLE
Reconsider use of "string pointers" in the Search API

### ISSUE THREAD

**Issue by bryevdv** [Azure/azure-sdk-for-python#11513]
Currently to specify a synonym map, one is created with a name, e.g. "test-syn-map":
```python
syn_map = client.create_synonym_map("test-syn-map", [
    "USA, United States, United States of America",
    "Washington, Wash. => WA",
])
```
and then later this name this passed to a `SearchableField`
```python
SearchableField(..., synonym_map="test-syn-map"),
```

This was confusing to some US study paticipants. It was not obvious that a parameter called `synonym_map` expected a string synonym map *name*. There are two potential ways to improve this:

### change the parameter name
Changing to `synonym_map_name` should make it more clear what is expected:
```python
SearchableField(..., synonym_map_name="test-syn-map"),
```

### accept objects instead
```python
SearchableField(..., synonym_map=syn_map),
```
and then the client code will extract and use the name in the API call automatically. 

---

There may be other places in the SDK where this improvement can be applied. 

cc @xiangyan99 @heaths

**Comment by heaths** [comment]
It would be harder for typed languages without unions (ahem, C#, I'm looking at you) to accept a string name or object, but this could be a place where other languages can do both. I do like the simplicity of  `synonym_map_name`, though. @bleroy , thoughs?

**Comment by xiangyan99** [comment]
Fixed in #11683
